joinfaces-autoconfigure/src/main/java/org/joinfaces/autoconfigure/omnifaces/OmnifacesAutoConfiguration.java

```diff
@@ -16,29 +16,69 @@
 
 package org.joinfaces.autoconfigure.omnifaces;
 
+import javax.servlet.ServletContainerInitializer;
+
 import org.joinfaces.autoconfigure.javaxfaces.JavaxFacesAutoConfiguration;
-import org.omnifaces.facesviews.FacesViewsInitializer;
+import org.joinfaces.autoconfigure.servlet.initializer.ServletContainerInitializerRegistrationBean;
+import org.omnifaces.facesviews.FacesViews;
 
 import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.web.servlet.ServletComponentScan;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
 /**
- * Spring Boot Auto Configuration of OmniFaces.
+ * {@link EnableAutoConfiguration Auto Configuration} for Omnifaces.
+ *
  * @author Marcelo Fernandes
  */
 @Configuration
+@ConditionalOnClass(FacesViews.class)
+@ConditionalOnWebApplication(type = ConditionalOnWebApplication.Type.SERVLET)
 @EnableConfigurationProperties(OmnifacesProperties.class)
-@ConditionalOnClass(FacesViewsInitializer.class)
 @AutoConfigureBefore(JavaxFacesAutoConfiguration.class)
-@ConditionalOnWebApplication(type = ConditionalOnWebApplication.Type.SERVLET)
+@ServletComponentScan("org.omnifaces")
 public class OmnifacesAutoConfiguration {
 
-	@Bean
-	public OmnifacesInitializerRegistrationBean omnifacesServletContainerInitializer() {
-		return new OmnifacesInitializerRegistrationBean();
+	/**
+	 * {@link EnableAutoConfiguration Auto Configuration} for Omnifaces 1.x.
+	 *
+	 * @author Lars Grefer
+	 */
+	@Configuration
+	@ConditionalOnClass(name = "org.omnifaces.facesviews.FacesViewsInitializer")
```

Comment:
@persapiens The `@ConditionalOnClass`-conditions of the two inner classes here can't be true an the same time, so one class will get 0% line-coverage. How should we handle this?

Reply:
I'll see it this week.

Reply:
I see two options here:

1. Allow the missing coverage
2. Fake something to get this two line somehow covered

I prefer the first option, because with the second option we would lie to the coverage tool and the whole coverage checking would become useless because we can't know how much coverage is created by _real_ tests.

Reply:
I prefer first option too.

Reply:
You configured maven-failsafe-plugin to work with mojarra and myfaces some time ago. Is it possible to image conditional execution to cover these possibilities?

Reply:
From maven's point of view mojarra and myfaces are two different dependencies, but omnifaces 1 and omnifaces 3 are the same dependency